src/core/annotation.js

```diff
@@ -704,6 +708,46 @@ class Annotation {
     }
   }
 
+  setRotation(mk) {
+    this.rotation = 0;
+    if (mk instanceof Dict) {
+      let angle = mk.get("R") || 0;
+      if (angle !== 0) {
+        // If the angle is negative then angle % 360 is still negative.
+        angle = (360 + (angle % 360)) % 360;
+        if (angle % 90 === 0) {
+          this.rotation = angle;
+        }
+      }
+    }
+  }
+
+  getRotationMatrix(annotationStorage) {
+    const storageEntry = annotationStorage
+      ? annotationStorage.get(this.data.id)
+      : undefined;
+    let rotation = storageEntry && storageEntry.rotation;
+    rotation = rotation === undefined ? this.rotation : rotation;
+
+    if (rotation === 0) {
+      return [1, 0, 0, 1, 0, 0];
+    }
+
+    const width = this.data.rect[2] - this.data.rect[0];
+    const height = this.data.rect[3] - this.data.rect[1];
+
+    switch (rotation) {
+      case 90:
+        return [0, 1, -1, 0, width, 0];
+      case 180:
+        return [-1, 0, 0, -1, width, height];
+      case 270:
+        return [0, -1, 1, 0, 0, height];
+      default:
+        return [1, 0, 0, 1, 0, 0];
```

Comment:
This should be unreachable, given the `if (rotation === 0) { ... }` block above, right?
Hence I'd suggest replacing this with:
```suggestion
        throw new Error("Invalid rotation");
```